website/blog/2022-11-31-simple-2.mdx

```diff
@@ -0,0 +1,115 @@
+---
+slug: simple-functions-2
+title: "Simple Functions: Efficient Complex Types"
+authors: [lsakka]
+tags: [tech-blog, simple-functions]
+---
+_This blogpost is part of a series of blog posts that discuss different features and optimizations of the simple function interface_.
+
+
+## Efficient Complex Types
+In this blogpost, we will discuss two major changes that we did recently to the simple function interface to make its performance comparable to the vector function implementations for functions that produce or consume complex types (Arrays, Maps and Rows).
+
+To show how much simpler simple functions are. The figure below shows a function _NestedMapSum_ that is written in both the simple and vector interfaces. The function consumes a nested map and computes the summations of all values and keys.
```

Comment:
maybe:

The figure below shows a function, _NestedMapSum_, written in both the simple and vector interfaces.